tsconfig.json

```diff
@@ -0,0 +1,68 @@
+{
+    "compilerOptions": {
+        /* Basic Options */
+        // "incremental": true,                   /* Enable incremental compilation */
+        "target": "es5" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
+        "module": "commonjs" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */,
```

Comment:
I am missing a point a little bit here with **module** option listed here and there with various configurations. Can you please elaborate on this one? What is the difference between all them and which one should we use. 
Or maybe you can send me one great article where I can look this all up altogether.

Reply:
https://medium.com/@tommedema/typescript-confusion-tsconfig-json-module-moduleresolution-target-lib-explained-65db2c44b491

to be short according to ts documentation "module" depends on "target" such that 
target === "ES3" or "ES5" ? "commonJS" : "ES6"